frontend/device/android/device.lua

```diff
@@ -400,21 +408,38 @@ function Device:showLightDialog()
     local action = android.lights.dialogState()
     if action == C.ALIGHTS_DIALOG_OK then
         self.powerd.fl_intensity = self.powerd:frontlightIntensityHW()
+        if self.powerd.fl_intensity > self.powerd.fl_min then
+            self.powerd.is_fl_on = true
+        else
+            self.powerd.is_fl_on = false
+        end
         logger.dbg("Dialog OK, brightness: " .. self.powerd.fl_intensity)
         if android.isWarmthDevice() then
             self.powerd.fl_warmth = self.powerd:getWarmth()
             logger.dbg("Dialog OK, warmth: " .. self.powerd.fl_warmth)
         end
-        local Event = require("ui/event")
-        local UIManager = require("ui/uimanager")
-        UIManager:broadcastEvent(Event:new("FrontlightStateChanged"))
+        return true
```

Comment:
The old way was fine as it was. You can change it to self.powerd:broadcast if you want

Reply:
OK